File: 1636-number-of-substrings-with-only-1s/number-of-substrings-with-only-1s.cpp
class Solution {
public:
    int numSub(string s) {
        int n = s.length();
        long long res = 0;
        int i = 0;
        while(i<n){
            if(s[i]=='1'){
                long long ct = 0;
                while(i<n && s[i]=='1'){
                    i++;
                    ct++;
                }
                res+= ct * (ct+1)/2;
            }
            else i++;
        }
        return (int) (res%1000000007);
    }
};

File: 1636-number-of-substrings-with-only-1s/number-of-substrings-with-only-1s.java
class Solution {
    public int numSub(String s) {
        int n = s.length();
        long res = 0;
        int i=0;
        while(i<n){
            if(s.charAt(i)=='1'){
                int ct = 0;
                int j = 0;
                for(j=i; j<n; j++){
                    if(s.charAt(j)=='1') ct++;
                    else break;
                }
                i=j;
                res+=(long)ct* (ct+1)/2;
            }
            else i++;
        }
        return (int)(res%1_000_000_007);
    }
}